manim/utils/decorators.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import inspect
+from typing import TYPE_CHECKING, TypeVar
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from typing_extensions import ParamSpec
+
+    P = ParamSpec("P")
+    T = TypeVar("T")
+
+
+def internal(f: Callable[P, T]) -> Callable[P, T]:
```

Comment:
```suggestion
F = TypeVar("F", bound=Callable[..., Any])

def internal(f: F, /) -> F:
```

Reply:
Is there a reason for this? According to the [mypy docs](https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators) It's better to use paramspec.

Reply:
`ParamSpec` is only useful when you explicitly have to deal with the decorated callable's parameters, e.g.:

```python
def printing_decorator(func: Callable[P, T]) -> Callable[P, T]:
    def wrapper(*args: P.args, **kwds: P.kwargs) -> T:
        print("Calling", func)
        return func(*args, **kwds)
    return wrapper
```

In your case, using a simple `TypeVar` bound to callables is easier to reason about. You are also certain that all metadata of the callable is retained (even though in pyright [it is also kept when using `ParamSpec`](https://github.com/microsoft/pyright/issues/7732#issuecomment-2067736165))